src/Symfony/Component/Routing/Matcher/RedirectableUrlMatcher.php

```diff
@@ -27,13 +28,17 @@ public function match($pathinfo)
         try {
             $parameters = parent::match($pathinfo);
         } catch (ResourceNotFoundException $e) {
-            if ('/' === substr($pathinfo, -1) || !in_array($this->context->getMethod(), array('HEAD', 'GET'))) {
+            if ('/' === substr($pathinfo, -1)) {
                 throw $e;
             }
 
             try {
                 parent::match($pathinfo.'/');
 
+                if (!in_array($this->context->getMethod(), array('HEAD', 'GET'))) {
+                    throw new MethodNotAllowedException(array('GET'));
```

Comment:
Same here, we don't know if GET is allowed at all.

Reply:
This cannot work because if don't know if GET for the route is allowed. We would need to make a similar check as in the PhpMatcherDumper like `in_array('GET', $route-getMethods())`. But we don't have access to the route, so that is difficult to do.
Example:
Route definition: POST /foo/
Request: POST /foo -> throws ResourceNotFoundException, then we check again with slash, i.e. POST /foo/
This works so we return MethodNotAllowedException with allow GET
But GET /foo is not allowed. It will just return a 404 again if someone tries that.

So I don't see how to implement this logic in the RedirectableUrlMatcher. We could just say, RedirectableUrlMatcher does not support that feature. And revert this here but keep the logic in PhpMatcherDumper which most people use anyway.